staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -649,9 +666,15 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		if err != nil {
 			return err
 		}
+
 		if r.ctx != nil {
+			r.ctx, _ = context.WithTimeout(r.ctx, r.requestTimeout)
```

Comment:
Why would you use a zero value timeout?  Seems like this should be gated.

Reply:
> Why would you use a zero value timeout? Seems like this should be gated.

Definitely needs gating. You'll be messing with behavior you don't own otherwise.  Is there a getter for the current timeout?  If not, you don't get to touch it after the context is set.